### PR TITLE
Leaf load failure

### PR DESCRIPTION
**Pull request by quadrismegistus**
Fix for https://github.com/Princeton-CDH/lenape-timetree/issues/132

### CODE REVIEW

**Review by rlskoeser — COMMENTED** [review]
Approach looks reasonable to me.
 
I'm seeing the error message visible in the footer when I run it locally, fwiw.

File: layouts/index.html [hunk 1]
```diff
@@ -28,11 +28,18 @@ <h1>{{ .Params.Title }}</h1>
...
+
   <div id="leaf-details">
     <article/>
   </div>
 </aside>
 
+<article id="leaferror" class="closed">
+  <header><h1>Uh oh</h1></header>
+  <div>Sorry buddy</div>
+</article>
```
[the remark this answers is not in the document]
Comment (quadrismegistus) [reply layouts/index.html:41]: An `h1` is not necessary -- just copying the intro article format. Here's a few mockups with/without header, etc:
Screen Shot 2023-04-18 at 11 23 59 AM
Screen Shot 2023-04-18 at 11 24 09 AM
Screen Shot 2023-04-18 at 11 24 21 AM

File: assets/js/leaves.js [hunk 2]
```diff
@@ -208,14 +223,22 @@ class Leaf {
...
-        panel.parentElement.classList.add("show-details");
-        panel.parentElement.classList.remove("closed");
         // store current leaf url in a data attribute so we can check for reload
         panel.dataset.showing = leafTarget.dataset.url;
+      })
+      .catch((response) => {
+        // clone error article and pass in to article
+        let errorArticle = document.querySelector("#leaferror").cloneNode(true);
+        errorArticle.classList.remove("closed");
+        errorArticle.classList.add("show-details");
```
[the remark this answers is not in the document]
Comment (quadrismegistus) [reply assets/js/leaves.js:233]: What if we add this to CSS:

```scss
// Hidden by default
#leaferror {
  visibility: hidden;
  
  &.showing {
    visibility: visible;
  }
}
```

And this to the python here:

```py
// clone error article and pass in to article
let errorArticle = document.querySelector("#leaferror").cloneNode(true);
errorArticle.classList.add("showing");
panel.querySelector("article").replaceWith(errorArticle);
```

File: assets/js/leaves.js [hunk 3]
```diff
@@ -208,14 +221,21 @@ class Leaf {
...
+        let errorArticle = document.querySelector("#leaferror").cloneNode(true);
+        panel.querySelector("article").replaceWith(errorArticle);
+        errorArticle.classList.add("showing");
       });
+
+    // scroll to the top, in case previous leaf was scrolled
+    panel.scrollTop = 0;
+    // make sure panel is active
+    panel.parentElement.classList.add("show-details");
+    panel.parentElement.classList.remove("closed");
```
Comment (rlskoeser) [inline assets/js/leaves.js:238]: I think these are redundant now, with my latest panel code changes in develop

Comment (quadrismegistus) [reply assets/js/leaves.js:238]: Do you mean we only need one of these, or neither, once the new changes from develop are in? Looks like I need only the first of these lines to create the behavior I want. Should I just merge develop into my PR now?

Comment (rlskoeser) [reply assets/js/leaves.js:238]: I think you shouldn't need them at all — opening and closing the panel should be handled in the panel code, leaf code shouldn't have to know about the container (ideally! but possibly not practically / not yet).

Yes, probably a good idea to update from develop and merge your changes with mine from the other branch (hopefully not to hard to resolve any conflicts)

**Comment by quadrismegistus** [comment]
Review notes/steps:

Right now any given leaf, upon selection, has a 50/50 chance of throwing an "Error loading the details" or successfully loading. Let's click around and make sure there are no surprises.

- [ ] The error message looks good when it shows
- [ ] The error message shows when it ought to

**Comment by gissoo** [comment]
@quadrismegistus thank you!

there is on problem I'm noticing: 
- the error on FF mobile emulator is showing below the footer – the footer is not showing up in other browsers' mobile emulator
Screenshot 2023-04-24 at 5 42 10 PM

**Pull request merged**